rekall/plugins/common/bovine.py

```diff
@@ -107,3 +113,100 @@ def render(self, renderer):
             utils.AttributedString(beer, beer_highlights),
             ("This is a fairly long line that shouldn't get wrapped.\n"
              "The same row has another line that shouldn't get wrapped."))
+        
+        renderer.section("Heatmap test:")
+        values = []
+        for digit in itertools.islice(colors.EulersDecimals(), 0xff):
+            values.append(dict(heat=float(digit + 1) * .1, value=digit))
+
+        randomized = visual_aides.Heatmap(
+            caption="Offset (p)",
+            # Some of the below xs stand for eXtreme. The other ones just
+            # look cool.
+            headers=["%0.2x" % x for x in xrange(0, 0xff, 0x10)],
+            row_headers=["0x%0.6x" % x for x
+                         in xrange(0x0, 0xfffff, 0x10000)],
+            values=values,
+            greyscale=False)
+        
+        gradual = visual_aides.Heatmap(
+            caption="Offset (v)",
+            headers=["%0.2x" % x for x in xrange(0, 0xff, 0x10)],
+            row_headers=["0x%0.6x" % x for x
+                         in xrange(0x0, 0xfffff, 0x10000)],
+            values=[dict(value="%x" % x, heat=x / 255.0) for x in xrange(256)],
+            greyscale=False)
+        
+        ranges_legend = visual_aides.MapLegend([
+            ("A", "kEfiACPIMemoryNVS", (0x00, 0xff, 0x00)),
+            ("A", "kEfiACPIReclaimMemory", (0xc7, 0xff, 0x50)),
+            ("Bc", "kEfiBootServicesCode", (0xff, 0xc7, 0x00)),
+            ("Bd", "kEfiBootServicesData", (0xff, 0x00, 0x00)),
+            ("M", "kEfiConventionalMemory", (0xff, 0xff, 0xff)),
+            ("Ec", "kEfiLoaderCode", (0x00, 0xff, 0xff)),
+            ("Ed", "kEfiLoaderData", (0x00, 0x00, 0xff)),
+            ("X", "kEfiReservedMemoryType", (0x00, 0x00, 0x00)),
+            ("Rc", "kEfiRuntimeServicesCode", (0xff, 0x00, 0xff)),
+            ("Rd", "kEfiRuntimeServicesData", (0xff, 0x00, 0x50))])
+        
+        ranges = visual_aides.Heatmap.from_memory_ranges(
+            caption="Offset (p)",
+            legend=ranges_legend,
+            ranges=[("kEfiConventionalMemory", 0x000000000000, 0x0000000a0000),
```

Comment:
you might consider putting test data into a test profile so it does not clutter the code.